.github/workflows/do-not-merge.yml

```
@@ -0,0 +1,17 @@
+name: Do Not Merge
+on:
+  pull_request:
+    types: [labeled, unlabeled]
+
+jobs:
+  do-not-merge:
+    # contains(search, item) is not case-sensitive
+    # Therefore 'Do Not Merge' and 'DO NOT MERGE' will also match
+    # see https://docs.github.com/en/actions/learn-github-actions/expressions#contains
+    if: contains(github.event.pull_request.labels.*.name, 'do not merge')
+    runs-on: ubuntu-latest
```

```suggestion
    name: Check
    runs-on: ubuntu-latest
```

This will change the name shown in the list of actions from `Do Not Merge / do-not-merge` to `Do Not Merge / Check`